InteractiveHtmlBom/web/render.js

```diff
@@ -356,13 +356,30 @@ function drawModules(canvas, layer, scalefactor, highlight) {
   }
 }
 
+var image_cache = {};
+
+function drawImage(ctx, d) {
```

Comment:
This function doesn't seem thread safe.
If another call to drawImage() comes in after first one finished but before image.onload completed, wouldn't this cause an attempt to draw unfinished Image object on the context?

I would just load the images in init sequence. There already is a safeguard that rendering is skipped until init is done.

Reply:
Thank you. I will move the code to the `init*()` then.